### PR TITLE
Added PR template

### PR DESCRIPTION
**Pull request by lucaswoj**
This PR adds a checklist which walks contributors through the essential steps of getting a PR 🚢-ready. 

This'll go hand-in-hand with #2957. 

Thoughts @mollymerp @springmeyer @mapsam @mourner @tmcw?

### CODE REVIEW

**Comment by springmeyer** [comment]
👍 

Minor edit: I would reword `To get your PR merged quickly` -> `To help your PR be most easily reviewed`

**Comment by lucaswoj** [comment]
Good idea!

**Comment by mollymerp** [comment]
Seems legittttt 🚢

**Comment by lucaswoj** [comment]
Thanks for the feedback @mollymerp @springmeyer. Looking forward to iterating on this in the future to make it perfect.
